metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1380,33 +1378,30 @@ def visit_join_to_time_spine_node(self, node: JoinToTimeSpineNode) -> SqlDataSet
         parent_data_set = node.parent_node.accept(self)
         parent_alias = self._next_unique_table_alias()
 
-        if node.use_custom_agg_time_dimension:
-            agg_time_dimension = node.requested_agg_time_dimension_specs[0]
-            agg_time_element_name = agg_time_dimension.element_name
-            agg_time_entity_links: Tuple[EntityReference, ...] = agg_time_dimension.entity_links
-        else:
-            agg_time_element_name = METRIC_TIME_ELEMENT_NAME
-            agg_time_entity_links = ()
+        agg_time_dimension_instances = parent_data_set.instances_for_time_dimensions(
+            node.requested_agg_time_dimension_specs
+        )
 
-        # Find the time dimension instances in the parent data set that match the one we want to join with.
-        agg_time_dimension_instances: List[TimeDimensionInstance] = []
-        for instance in parent_data_set.instance_set.time_dimension_instances:
-            if (
-                instance.spec.date_part is None  # Ensure we don't join using an instance with date part
-                and instance.spec.element_name == agg_time_element_name
-                and instance.spec.entity_links == agg_time_entity_links
-            ):
-                agg_time_dimension_instances.append(instance)
+        # Select the dimension for the join from the parent node because it may not have been included in the request.
+        # Default to using metric_time for the join if it was requested, otherwise use the agg_time_dimension.
```

Comment:
This is the one place we still have some differentiation between `metric_time` and `agg_time`. If you request both, we have to choose one of them to use as the join column, and we default to using `metric_time`.